### PR TITLE
using report name value to check archived

### PR DESCRIPTION
**Pull request by DylanDylann**
### Details

We need to pass in reportNameValuePairs wherever [isArchivedRoom](https://github.com/Expensify/App/blob/2e7e212ea7ff50317e85a94d7676d2a419f8b0f2/src/libs/ReportUtils.ts#L1256-L1262) is called 

### Fixed Issues

$ https://github.com/Expensify/App/issues/45323
PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot 2024-07-15 at 17 01 12

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot 2024-07-15 at 17 00 35

</details>

<details>
<summary>iOS: Native</summary>

Screenshot 2024-07-15 at 17 00 09

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/d23370d7-e76e-4f7b-8c87-476f226b34a3

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/5f55208e-844a-4622-bee3-ccb87a05d2bd

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-07-15 at 17 00 47

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@Ollyws This PR isn't required the C+ reviewer. Let's ignore

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:1292]: Also include https://github.com/Expensify/App/pull/45320 in here

Comment (srikarparsi) [reply src/libs/ReportUtils.ts:1292]: Cool, we can just use private_isArchived here since this [PR](https://github.com/Expensify/App/pull/45320/files) has been merged

**Review by srikarparsi — CHANGES_REQUESTED** [review]
Can we also refactor [ReportScreen](https://github.com/Expensify/App/blob/77d95c54b3b4d9d9eb7db560f150137de433f657/src/pages/home/ReportScreen.tsx#L818) and ReportFooter to use getReportNameValuePair

File: src/libs/ReportUtils.ts
Comment (srikarparsi) [inline src/libs/ReportUtils.ts:641]: Is there a reason you called this `getReportNameValuePair` instead of `getReportNameValuePairs`? (With the s) Otherwise, I think it's better to call this `getReportNameValuePairs`

**Review by srikarparsi — CHANGES_REQUESTED** [review]

**Comment by srikarparsi** [comment]
Hey @DylanDylann, thanks for the quick work. I wanted to wait until this [PR](https://github.com/Expensify/App/pull/45320/files) got merged before merging this one but it's merged now. I left a couple comments above and there's also merge conflicts now since I waited. Once you get to those, please @ me and I can do a final review + merge.

**Comment by DylanDylann** [comment]
@srikarparsi 

> Can we also refactor [ReportScreen](https://github.com/Expensify/App/blob/77d95c54b3b4d9d9eb7db560f150137de433f657/src/pages/home/ReportScreen.tsx#L818) and ReportFooter to use getReportNameValuePair

You mean that we should use ReportUtils.getReportNameValuePairs function in ReportScreen and ReportFooter

**Comment by DylanDylann** [comment]
ReportUtils.getReportNameValuePairs function shouldn't be used in the component directly. In component, we should get reportNameValuePair from Onyx

**Comment by srikarparsi** [comment]
> ReportUtils.getReportNameValuePairs function shouldn't be used in the component directly. In component, we should get reportNameValuePair from Onyx

Sorry, I meant useOnyx, similar to what you have done in this PR

Also, the code looks good. Do you think you could add test steps and screenshots just so that if there are regressions later on then we can easily see if they came from this PR or not? They can be something on the lines of creating a workspace, deleting it, and making sure the report shows up as archived. Once that's done I think this is good to merge

**Comment by DylanDylann** [comment]
> Sorry, I meant useOnyx, similar to what you have done in this PR. It is currently using withOnyx.

Do you mean we should move all Onyx keys to `useOnyx`? In this PR, if that file currently is using withOnyx, I still keep it

**Comment by srikarparsi** [comment]
> But I am fine to add some test cases here

Yes, that would be great

> Do you mean we should move all Onyx keys to useOnyx?

Just the reportNameValuePairs keys. I believe [this](https://github.com/Expensify/App/blob/39463bdccefd9017b3f4beea889e57e58ad5644e/src/pages/home/ReportScreen.tsx#L819) is the only place we're using withOnyx for the reportNameValuePairs key so if we can refactor that to use useOnyx that would be great. I also think we can move the useOnyx into ReportFooter since we don't need the reportNameValuePairs object in ReportScreen.

**Comment by DylanDylann** [comment]
@srikarparsi All done

**Comment by srikarparsi** [comment]
Awesome thanks, one quick bug I spotted (the space between the last message and the archived box is too large):

Expected:
Say hello!

Actual:
Say hello!

**Review by suneox — APPROVED** [review]
Overall, the changes look good to me. The condition check `isArchivedRoom` still keeps checking both `statusNum` and `stateNum`, so it should work for both the legacy and new approaches. My test results are included in the Screenshots/Videos section

**Comment by srikarparsi** [comment]
Cool, going to go ahead and merge to avoid conflicts but let's fix this [spacing issue](https://github.com/Expensify/App/pull/45330#issuecomment-2228930851) as a follow up

**Review by srikarparsi — APPROVED** [review]

**Pull request merged**
